### PR TITLE
Random sampling in GeneralLinear

### PR DESCRIPTION
**Pull request by pchauchat**
Random sampling based on the fact that a random matrix is invertible with probability 1. Similarly to what is done for the hypersphere, a check on the determinant is made to have a numerically acceptable matrix.

The value of the determinant threshold is completely arbitrary, and is surely not the best one. I guess it should vary with `n`.

### CODE REVIEW

**Comment by pchauchat** [comment]
About the remaining DeepSource error, I am conflicted: the `belongs` in `matrices` does not need to do any check, and therefore does not need any tolerance value. To have matching signatures would mean add an unused parameter to one, or remove a potentially useful parameter to the other (unless it is specified in the core of the function).
Any preferrences @ninamiolane @nkoep  ?

File: geomstats/geometry/general_linear.py [hunk 1]
```diff
@@ -30,6 +35,33 @@ def inv(point):
         """Return the inverse of a matrix."""
         return gs.linalg.inv(point)
 
+    def random_uniform(self, n_samples=1, tol=1e-8):
+        """Sample in GL(n) from the uniform distribution.
+
+        Parameters
+        ----------
+        n_samples: int, optional
```
Comment (ninamiolane) [inline geomstats/geometry/general_linear.py:43]: super-NIT: "n_samples : "
-> weirdly, there needs to be a space before the : in numpy's conventions for docstrings, which we are following.

File: geomstats/geometry/general_linear.py [hunk 2]
```diff
@@ -30,6 +35,33 @@ def inv(point):
...
+            Number of samples.
+        tol: float, optional
```
Comment (ninamiolane) [inline geomstats/geometry/general_linear.py:45]: Ditto.

File: geomstats/geometry/general_linear.py [hunk 3]
```diff
@@ -30,6 +35,33 @@ def inv(point):
         """Return the inverse of a matrix."""
         return gs.linalg.inv(point)
 
+    def random_uniform(self, n_samples=1, tol=1e-8):
```
Comment (ninamiolane) [inline geomstats/geometry/general_linear.py:38]: Numpy uses 1e-6 as a tolerance default, for example in np.isclose. You could use the same?

File: geomstats/geometry/general_linear.py [hunk 4]
```diff
@@ -30,6 +35,33 @@ def inv(point):
...
+        samples = gs.random.rand(n_samples, self.n, self.n)
+        while True:
+            dets = gs.linalg.det(samples)
+            indcs = gs.isclose(dets, 0.0, atol=tol)
+            num_bad_samples = gs.sum(indcs)
+            if num_bad_samples == 0:
+                break
+            samples[indcs, :] = gs.random.rand(num_bad_samples, self.n, self.n)
+
+        return samples
```
Comment (ninamiolane) [inline geomstats/geometry/general_linear.py:63]: Great, thanks! Could you squeeze axis 0, for when there is only one sample?

**Comment by ninamiolane** [comment]
@pchauchat (In addition to the PR's comments,) could you solve the GitHub conflicts? Thnaks!

**Comment by pchauchat** [comment]
It's getting confused, I redid it in #558

**Pull request closed without merging**